app/main/views/services.py

```diff
@@ -715,41 +722,80 @@ def set_remove_last_subsection_error_message():
                 ))
 
 
-@main.route('/frameworks/<framework_slug>/submissions/<lot_slug>/previous-services', methods=['GET'])
+@main.route('/frameworks/<framework_slug>/submissions/<lot_slug>/previous-services', methods=['GET', 'POST'])
 @login_required
-def list_previous_services(framework_slug, lot_slug):
+def previous_services(framework_slug, lot_slug):
     framework, lot = get_framework_and_lot_or_404(data_api_client, framework_slug, lot_slug)
     if framework['status'] != 'open':
         abort(404)
 
-    copy_all = request.args.get('copy_all', None)
+    form = OneServiceLimitCopyServiceForm(lot['name'].lower()) if lot.get('oneServiceLimit') else None
     source_framework_slug = content_loader.get_metadata(framework['slug'], 'copy_services', 'source_framework')
-    source_framework = get_framework_or_404(data_api_client, source_framework_slug)
 
-    previous_services = data_api_client.find_services(
+    try:
+        source_framework = data_api_client.get_framework(source_framework_slug)['frameworks']
```

Comment:
Maybe a `get_framework_or_500`?

Would feel like a weird method name but could be explained by a docstring and would remove this duplication.